bip-0360.mediawiki

```diff
@@ -1,724 +1,452 @@
+* Let ''q'' be the 32-byte array containing the witness program (the second push in the scriptPubKey) which represents the root of the tapleaf Merkle tree.
 * Fail if the witness stack does not have two or more elements.
-* If there are at least three witness elements, and the first byte of the last element is 0x50, this last element is called ''annex'' ''a'' and is removed from the witness stack. The annex (or the lack thereof) is always covered by the signature and contributes to transaction weight, but is otherwise ignored during Taproot validation.
+* If there are at least three witness elements, and the first byte of the last element is 0x50, this last element is called ''annex'' ''a'' and is removed from the witness stack. The annex (or the lack thereof) is always covered by the signature and contributes to transaction weight, but is otherwise ignored during P2TSH validation.
 * There must be at least two witness elements left.
-** Call the second-to-last stack element ''s'', the script (as defined in [[bip-0341.mediawiki|BIP 341]])
+** Call the second-to-last stack element ''s'', the script (as defined in [https://github.com/cryptoquick/bips/blob/p2qrh/bip-0341.mediawiki BIP 341])
 ** The last stack element is called the control block ''c'', and must have length ''1 + 32 * m'', for a value of ''m'' that is an integer between 0 and 128, inclusive. Fail if it does not have such a length.
-** Let ''v = c[0] & 0xfe'' be the ''leaf version'' (as defined in [[bip-0341.mediawiki|BIP 341]]). To maintain ''leaf version'' encoding compatibility the last bit of c[0] is unused and must be 1 <ref>'''Why set the last bit of c[0] to one?''' Consider a faulty implementation that deserializes the ''leaf version'' as c[0] rather than c[0] & 0xfe for both P2TR and P2QRH. If they test against P2QRH outputs and require that last bit is 1, this deserialization bug will cause an immediate error.</ref>.
-** Let ''k<sub>0</sub> = hash<sub>TapLeaf</sub>(v || compact_size(size of s) || s)''; also call it the ''tapleaf hash''.
-** For ''j'' in ''[0,1,...,m-1]'':
-*** Let ''e<sub>j</sub> = c[33+32j:65+32j]''.
-*** Let ''k<sub>j+1</sub> depend on whether ''k<sub>j</sub> < e<sub>j</sub>'' (lexicographically):
-**** If ''k<sub>j</sub> < e<sub>j</sub>'': ''k<sub>j+1</sub> = hash<sub>TapBranch</sub>(k<sub>j</sub> || e<sub>j</sub>)''.
-**** If ''k<sub>j</sub> &ge; e<sub>j</sub>'': ''k<sub>j+1</sub> = hash<sub>TapBranch</sub>(e<sub>j</sub> || k<sub>j</sub>)''.
-** Let ''r = hash<sub>QuantumRoot</sub>(k<sub>m</sub>)''.
-** If ''q &ne; r'', fail.
+** Let ''v = c[0] & 0xfe'' be the ''leaf version'' (as defined in [https://github.com/cryptoquick/bips/blob/p2qrh/bip-0341.mediawiki BIP 341]). To maintain ''leaf version'' encoding compatibility the last bit of c[0] is unused and must be 1 [<nowiki/>[https://github.com/cryptoquick/bips/blob/p2qrh/bip-0360.mediawiki#cite_note-12 6]].
+** Let ''k0 = hashTapLeaf(v || compact_size(size of s) || s)''; also call it the ''tapleaf hash''.
```

Comment:
It looks like someone went in an removed the subscript formatting. Why? How is this better?